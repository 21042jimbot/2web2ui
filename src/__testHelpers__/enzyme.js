export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
