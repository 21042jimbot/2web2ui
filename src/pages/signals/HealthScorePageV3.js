import React, { useCallback } from 'react';
import { Link } from 'react-router-dom';
import { selectHealthScoreDetailsV3 } from 'src/selectors/signals';
import Page from './components/SignalsPage';
import { Grid, Panel } from '@sparkpost/matchbox';
import { Button } from 'src/components/matchbox';
import withDetails from './containers/withDetails';
import BarChart from './components/charts/barchart/BarChart';
import withDateSelection from './containers/withDateSelection';
import { getHealthScore } from 'src/actions/signals';
import TooltipMetric from 'src/components/charts/TooltipMetric';
import thresholds from './constants/healthScoreThresholds';
import { roundToPlaces } from 'src/helpers/units';
import moment from 'moment';
import HealthScoreMetric from './components/HealthScoreMetric';
import DateFilter from './components/filters/DateFilter';
import HealthScoreLegend from './components/HealthScoreLegend';
import Callout from 'src/components/callout/Callout';
import { Loading } from 'src/components';

export function HealthScorePageV3(props) {
  const {
    data = [],
    empty,
    error,
    facet,
    facetId,
    handleDateHover,
    handleDateSelect,
    hoveredDate,
    gap,
    loading,
    resetDateHover,
    selectedDate,
    subaccountId,
    xTicks,
  } = props;

  const getXAxisProps = useCallback(
    () => ({
      ticks: xTicks,
      tickFormatter: tick => moment(tick).format('M/D'),
    }),
    [xTicks],
  );

  const renderChart = () => {
    if (empty) {
      return <Callout title="No Data Available">Insufficient data to populate this chart</Callout>;
    }

    if (error) {
      return <Callout title="Unable to Load Data">{error.message}</Callout>;
    }

    if (loading) {
      return (
        <div style={{ height: '220px', position: 'relative' }}>
          <Loading />
        </div>
      );
    }
    return (
      <BarChart
        gap={gap}
        onClick={handleDateSelect}
        onMouseOver={handleDateHover}
        onMouseOut={resetDateHover}
        disableHover={false}
        selected={selectedDate}
        hovered={hoveredDate}
        timeSeries={data}
        tooltipContent={({ payload = {} }) =>
          payload.ranking && (
            <TooltipMetric
              label="Health Score"
              color={thresholds[payload.ranking].color}
              value={`${roundToPlaces(payload.health_score * 100, 1)}`}
            />
          )
        }
        yAxisRefLines={[
          { y: 0.8, stroke: thresholds.good.color, strokeWidth: 1 },
          { y: 0.55, stroke: thresholds.danger.color, strokeWidth: 1 },
        ]}
        yKey="health_score"
        yAxisProps={{
          ticks: [0, 0.55, 0.8, 1],
          tickFormatter: tick => parseInt(tick * 100),
        }}
        xAxisProps={getXAxisProps()}
      />
    );
  };

  const scoreMetricDate = hoveredDate || selectedDate;
  const { injections, health_score = 0 } =
    data.find(dataPoint => dataPoint.date === scoreMetricDate) || {};
  return (
    <Page
      breadcrumbAction={{
        content: 'Back to Health Score Overview',
        to: '/signals/health-score',
        component: Link,
      }}
      title="Health Score"
      facet={facet}
      facetId={facetId}
      subaccountId={subaccountId}
      primaryArea={
        <Button primary component={Link} to="/alerts/create">
          Create Alert
        </Button>
      }
    >
      <Panel>
        <Panel.Section>
          <Grid>
            <Grid.Column xs={8}>
              {!loading && !empty && !error && (
                <HealthScoreMetric
                  date={scoreMetricDate}
                  injections={injections}
                  score={health_score * 100}
                />
              )}
            </Grid.Column>
            <Grid.Column xs={4}>
              <DateFilter left />
              <HealthScoreLegend />
            </Grid.Column>
          </Grid>
          {renderChart()}
        </Panel.Section>
      </Panel>
    </Page>
  );
}

export default withDetails(
  withDateSelection(HealthScorePageV3),
  { getHealthScore },
  selectHealthScoreDetailsV3,
);
