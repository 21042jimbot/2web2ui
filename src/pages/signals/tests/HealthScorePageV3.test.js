import { shallow } from 'enzyme';
import React from 'react';
import { HealthScorePageV3 } from '../HealthScorePageV3';

describe('Signals Health Score Page V3', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      facetId: 'test.com',
      facet: 'sending-domain'
    };
    wrapper = shallow(<HealthScorePageV3 {...props}/>);
  });

  it('renders correctly when receiving data', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
