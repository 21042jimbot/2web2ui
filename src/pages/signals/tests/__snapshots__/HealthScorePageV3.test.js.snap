// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signals Health Score Page V3 renders correctly when receiving data 1`] = `
<SignalsPage
  breadcrumbAction={
    Object {
      "component": [Function],
      "content": "Back to Health Score Overview",
      "to": "/signals/health-score",
    }
  }
  facet="sending-domain"
  facetId="test.com"
  primaryArea={
    <Button
      component={[Function]}
      primary={true}
      size="default"
      to="/alerts/create"
    >
      Create Alert
    </Button>
  }
  title="Health Score"
>
  <Panel>
    <Panel.Section>
      <Grid>
        <Grid.Column
          xs={8}
        >
          <HealthScoreMetric
            score={NaN}
          />
        </Grid.Column>
        <Grid.Column
          xs={4}
        >
          <Connect(DateFilter)
            left={true}
          />
          <HealthScoreLegend />
        </Grid.Column>
      </Grid>
      <BarChart
        activeFill="#22838A"
        disableHover={false}
        fill="#B3ECEF"
        gap={1}
        height={250}
        margin={
          Object {
            "bottom": 5,
            "left": 18,
            "right": 8,
            "top": 21,
          }
        }
        timeSeries={Array []}
        tooltipContent={[Function]}
        width="99%"
        xAxisProps={
          Object {
            "tickFormatter": [Function],
            "ticks": undefined,
          }
        }
        xAxisRefLines={
          Array [
            Object {
              "label": Object {
                "className": "xDividerLabel",
                "offset": 14,
                "position": "top",
                "value": "New",
              },
              "stroke": "#d2d2d7",
              "strokeWidth": 1,
              "x": "2019-06-07",
            },
            Object {
              "label": Object {
                "className": "xDividerLabel",
                "position": "top",
                "value": "Model",
              },
              "strokeWidth": 0,
              "x": "2019-06-07",
            },
          ]
        }
        xKey="date"
        yAxisProps={
          Object {
            "tickFormatter": [Function],
            "ticks": Array [
              0,
              0.55,
              0.8,
              1,
            ],
          }
        }
        yAxisRefLines={
          Array [
            Object {
              "stroke": "#2CC995",
              "strokeWidth": 1,
              "y": 0.8,
            },
            Object {
              "stroke": "#FF594D",
              "strokeWidth": 1,
              "y": 0.55,
            },
          ]
        }
        yKey="health_score"
        yRange={
          Array [
            "auto",
            "auto",
          ]
        }
      />
    </Panel.Section>
  </Panel>
</SignalsPage>
`;
